taxonomy-editor-frontend/src/pages/root-nodes/index.tsx

```diff
@@ -0,0 +1,207 @@
+import EditIcon from "@mui/icons-material/Edit";
+import AddBoxIcon from "@mui/icons-material/AddBox";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+
+import CreateNodeDialogContent from "./CreateNodeDialogContent";
+import useFetch from "../../components/useFetch";
+import { createBaseURL } from "../editentry/createURL";
+import { toTitleCase } from "../../utils";
+import { greyHexCode } from "../../constants";
+
+type DataType = {
+  id: string;
+  string: string | Array<string>;
+};
+
+type Props = {
+  addNavLinks: ({
+    taxonomyName,
+    branchName,
+  }: {
+    taxonomyName: string;
+    branchName: string;
+  }) => void;
+};
+
+const RootNodes = ({ addNavLinks }: Props) => {
+  const [openCreateNodeDialog, setOpenCreateNodeDialog] = useState(false);
+  const [openCreateNodeSuccessSnackbar, setCreateNodeOpenSuccessSnackbar] =
+    useState(false);
+
+  const { taxonomyName, branchName } = useParams();
+  const baseUrl = createBaseURL(taxonomyName, branchName);
+
+  const {
+    data: nodes,
+    isPending,
+    isError,
+    errorMessage,
+  } = useFetch<Array<DataType[]>>(`${baseUrl}rootentries`);
+
+  useEffect(
+    function defineMainNavLinks() {
+      if (!branchName || !taxonomyName) return;
+
+      addNavLinks({ branchName, taxonomyName });
+    },
+    [taxonomyName, branchName, addNavLinks]
+  );
+
+  const handleCloseAddDialog = () => {
+    setOpenCreateNodeDialog(false);
+  };
+
+  const handleCloseSuccessSnackbar = () => {
+    setCreateNodeOpenSuccessSnackbar(false);
+  };
+
+  if (isError || !branchName || !taxonomyName) {
+    return (
+      <Box>
+        <Typography variant="h5">{errorMessage}</Typography>
+      </Box>
+    );
+  }
+
+  if (isPending || !nodes) {
+    return (
+      <Box>
+        <Typography variant="h5">Loading...</Typography>
+      </Box>
+    );
+  }
+
+  return (
+    <>
+      <div>
+        <Typography sx={{ mb: 2, mt: 2, ml: 2 }} variant="h4">
+          Root Nodes:
+        </Typography>
+
+        <TableContainer sx={{ ml: 2, width: 375 }}>
+          <Table style={{ border: "solid", borderWidth: 1.5 }}>
+            <TableHead>
+              <TableCell align="left">
+                <Typography variant="h6">Taxonony Name</Typography>
+              </TableCell>
+              <TableCell align="left">
+                <Typography variant="h6">Branch Name</Typography>
+              </TableCell>
+            </TableHead>
+            <TableBody>
+              <TableCell align="left" component="td" scope="row">
+                <Typography variant="body1">
+                  {toTitleCase(taxonomyName ?? "")}
+                </Typography>
+              </TableCell>
+              <TableCell align="left" component="td" scope="row">
+                <Typography variant="body1">{branchName}</Typography>
+              </TableCell>
+            </TableBody>
+          </Table>
+        </TableContainer>
```

Comment:
If possible, can we add a bottom border for this table component? It seems out of place without it.